src/main/SettingsUpgradeUtils.cpp

```diff
@@ -160,6 +160,15 @@ validateConfigUpgradeSet(ConfigUpgradeSet const& upgradeSet)
                 throw std::runtime_error("Invalid contractLedgerCost");
             }
         }
+        else if (entry.configSettingID() == CONFIG_SETTING_STATE_ARCHIVAL)
+        {
+            // 1048576 is our current phase1 scan size, and we don't expect to
+            // go past this anytime soon.
+            if (entry.stateArchivalSettings().evictionScanSize > 1048576)
```

Comment:
This seems a bit too restrictive to me, why not, say, at least 10 MB? IIUC it's easy to update this code, but OTOH it would be better if we could just use a slightly higher value if necessary without creating a new build.

Reply:
10 MB is actually not safe, which is why we're adding this limit. This is also just a util used locally, so if we wanted to go above 1048576, we could just update the code, run make, and then run the command.